### PR TITLE
Add metric for failed and succeeded repair tasks per table

### LINKED ISSUE #295: Metrics on number of attempts to repair
When a range isn't able to repair, it will create alarms after a configurable amount of time. However, it would be good to be able to measure and track the number of failed tries that have been attempted on a particular range. This could be used to preemptively seek out problems before they get worse.

### PR DESCRIPTION
**Pull request by masokol**
The new metric is a sliding window metric, counting average failed repair sessions the last 1 minute, 5 minutes and 15 minutes.
Closes #295

### CODE REVIEW

**Review by tommystendahl — APPROVED** [review]

**Review by itskarlsson — CHANGES_REQUESTED** [review]

File: docs/METRICS.md [hunk 1]
```diff
@@ -150,6 +150,10 @@ Usually the RepairFailedTime should be all zeros but if it's not the reason can
 
     The effective remaining repair time (in milliseconds) for the table to be fully repaired (time ecChronos waits for cassandra to perform repair).
 
+* RepairFailedAttempts
+
+    The amount of repair sessions that have failed for the repair job. The metric is reset each time repair job runs.
```
[the remark this answers is not in the document]
Comment (masokol) [reply docs/METRICS.md:155]: As discussed, we would like to have failed repairs over time (rate like).
For this we need dropwizard metrics 4.X and most probably cassandra driver 4.X.

From my local testing, stepping driver major version will be a big task so this PR will have to wait until then.

**Review by tommystendahl — APPROVED** [review]

File: docs/METRICS.md [hunk 2]
```diff
@@ -183,4 +187,17 @@ If this value is beyond the alarm intervals an alarm should have been sent.
 The value represents the effective remaining repair time for the table to be fully repaired in milliseconds.
 This is the time ecChronos will have to wait for Cassandra to perform repair,
 this is an estimation based on the last repair of the table.
-The value should be `0` if there is no repair ongoing for this table.
\ No newline at end of file
+The value should be `0` if there is no repair ongoing for this table.
+
+| t          | count | mean_rate | m1_rate  | m5_rate  | m15_rate | rate_unit     |
+|------------|-------|-----------|----------|----------|----------|---------------|
+| 1660042648 | 0     | 0.000000  | 0.000000 | 0.000000 | 0.000000 | events/second |
```
Comment (itskarlsson) [inline docs/METRICS.md:194]: This is valuable information. Perhaps we should also add a counter for succeeded RepairTasks.

File: docs/METRICS.md [hunk 3]
```diff
@@ -183,4 +193,35 @@ If this value is beyond the alarm intervals an alarm should have been sent.
...
+\<keyspace\>.\<table\>-\<table-id\>-SucceededRepairTasks
+
+The count represents the total amount of failed/succeeded repair tasks for the table.
+The mean rate is the rate at which events have occurred since the beginning.
+The `m1_rate`, `m5_rate` and `m_15_rate` are the rates at which events have occurred for the past 1 minute,
+5 minutes and 15 minutes.
+For example an `m1_rate` of `15` would mean that 15 events have occurred in the past minute.
+
+For `\<keyspace\>.\<table\>-\<table-id\>-FailedRepairTasks`, `m1_rate`, `m5_rate` and `m15_rate` should always be `0`
+in healthy clusters (assuming repairs are actually running).
```
Comment (itskarlsson) [inline docs/METRICS.md:218]: "(assuming repairs are actually running)"

Should be 0 even if repairs are not running

Something like:
" should be close to `0` in healthy clusters. A positive rate might indicate a problem and warrant a look into the logs."

File: docs/METRICS.md [hunk 4]
```diff
@@ -183,4 +193,35 @@ If this value is beyond the alarm intervals an alarm should have been sent.
...
+If repairs are not running the `m1_rate`, `m5_rate` and `m15_rate` will also show `0`.
```
Comment (itskarlsson) [inline docs/METRICS.md:219]: will show '0'.

**Review by itskarlsson — APPROVED** [review]

**Pull request merged**